ApollosProject/apollos-prototype
# Adds react-native-config instead of react-native-dotenv

## DESCRIPTION

### What does this PR do, or why is it needed?

While `react-native-dotenv` is great because it doesn't rely on native code, it's comes with it's own set of pitfalls, including
- No obvious way to flush cached values (impervious to `yarn nuke`, only way to flush is make arbitrary change to place value is imported.
- Very brittle. Referencing values in the app that don't exist in a specific dotenv have a tendency to break builds in uexpected ways. 

`react-native-config` requires native code, but is less brittle and problematic than `react-native-dotenv`. It doesn't require a rebuild to change `.env` values, but rebuilds _do_ update the imported values. 

### What design trade-offs/decisions were made?

n/a 

### How do I test this PR?

1. `yarn`
2. Build iOS, ensure app loads.
3. Build android, ensure app loads.

### What automated tests did you write?
n/a, updated / removed mock files.

## TODO:

- [x] Closes #483 

## REVIEW:

**Manual QA**


**Code Review: Questions to consider**


> The purpose of PR Review is to _improve the quality of the software._
